### PR TITLE
Also install a code gallery page if not found during cmake time.

### PR DESCRIPTION
**Pull request by bangerth**
This makes sure that we always have a doxygen target to link to available. The target
page may simply say that the code gallery was not available.

In reference to #1314.

### CODE REVIEW

File: doc/doxygen/code-gallery/CMakeLists.txt [hunk 1]
```diff
@@ -21,6 +21,9 @@
 
 ADD_CUSTOM_TARGET(code-gallery)
 
+MESSAGE(STATUS "Setting up code gallery documentation from ${DEAL_II_CODE_GALLERY_DIRECTORY}.")
```
Comment (tjhei) [inline doc/doxygen/code-gallery/CMakeLists.txt]: should this message happen below the ```SET_IF_EMPTY``` call?

Comment (tjhei) [inline doc/doxygen/code-gallery/CMakeLists.txt]: On second thought, I would move it back into the IF. Any good reason you moved it outside?

Comment (bangerth) [inline doc/doxygen/code-gallery/CMakeLists.txt]: I always set it up, but the code gallery page is sometimes empty and only shows a terse message that there are no programs to show.

Comment (bangerth) [inline doc/doxygen/code-gallery/CMakeLists.txt]: I don't care either way. It matches line 134.

**Comment by bangerth** [comment]
@tjhei -- I've made the change you requested.

**Pull request merged**